src/lsm/compaction.zig

```diff
@@ -1561,13 +1562,22 @@ pub fn CompactionType(
             compaction.level_b_position.value += merge_result.consumed_b;
             compaction.table_builder.value_count += merge_result.produced;
 
+            inline for (.{
+                compaction.level_a_position.value,
+                compaction.level_b_position.value,
+                compaction.table_builder.value_count,
+            }) |value_count| {
+                assert(value_count <= Table.data.value_count_max);
+            }
+
             if (compaction.table_info_a.? == .immutable) {
                 compaction.counters.in += merge_result.consumed_a;
             }
             compaction.counters.dropped += merge_result.dropped;
 
             compaction.quotas.done += merge_result.consumed_a + merge_result.consumed_b;
-            assert(compaction.table_builder.value_count <= Table.data.value_count_max);
+            assert(compaction.quotas.done <= compaction.quotas.bar);
+            assert(compaction.quotas.done <= compaction.quotas.beat + Table.data.value_count_max);
```

Comment:
This one is quite load beading actually! It proves our grid reservation calculation in beat_commence to be correct!

Reply:
Ah, no, this is incorrect! We could arbitrary overshoot our beat quota in reality! This happens, for example, if level an and level b perfectly annihilate each other through tombstones, producing no output. In this case we'd have to run the entire companion in a single beat. 

This doesn't violate our gird reservation assumption --- although we consume _more_ values then expected, we don't produce more!